File: kakao/19년 겨울 인턴/징검다리.cpp
#include <string>
#include <vector>
using namespace std;

bool possi(int n,vector<int> stones,int k)
{
    int cnt=0;
    for(int i=0;i<stones.size();i++)
    {
        if(stones[i]<=n)
            cnt++;
        else
            cnt=0;
        if(cnt>=k)
            return false;
    }
    return true;
}

int solution(vector<int> stones,int k)
{
    int hi=200000000;
    int lo=0;
    while(lo<=hi)
    {
        int mid=(hi+lo)/2;
        if(possi(mid,stones,k))
            lo=mid+1;
        else
            hi=mid-1;
    }
    return lo;
}

File: kakao/19년 겨울 인턴/징검다리.py
def check(stones,k,n):
    temp=0
    for stone in stones:
        if stone<=n:
            temp+=1
        else:
            temp=0
        if temp>=k:
            return False
    return True

def solution(stones,k):
    l=0
    r=200000000
    while l<=r:
        n=(l+r)//2
        if check(stones,k,n):
            l=n+1
        else:
            r=n-1
    return l